src/hotspot/share/opto/node.cpp

```diff
@@ -894,33 +894,26 @@ int Node::replace_edges_in_range(Node* old, Node* neww, int start, int end) {
 
 //-------------------------disconnect_inputs-----------------------------------
 // NULL out all inputs to eliminate incoming Def-Use edges.
-// Return the number of edges between 'n' and 'this'
-int Node::disconnect_inputs(Node *n, Compile* C) {
-  int edges_to_n = 0;
-
+void Node::disconnect_inputs(Compile* C) {
   uint cnt = req();
+
   for( uint i = 0; i < cnt; ++i ) {
-    if( in(i) == 0 ) continue;
-    if( in(i) == n ) ++edges_to_n;
-    set_req(i, NULL);
+    if( in(i) == nullptr ) continue;
```

Comment:
Perhaps a matter of preference, but this could now be simplified to:
```
if (in(i) != nullptr) {
  set_req(i, nullptr);
}
```

Reply:
Agree with suggestion. It is HS code style requirements.